Meetups/26/2025-03-27-minutes.md

```diff
@@ -0,0 +1,186 @@
+
+Henk: All messages have an envelope, like message id, timestamp etc.
+
+Henk: You can actually use the bindings independent from the hub.
+
+Henk: Current services are: Launcher, history of events, web ui, server CA management, and more.
+
+Henk: Protocol bindings include zwave, insteon, 1-wire
+
+Henk: The command line has some utilities. You can generate client code and also vocabularies.
+
+Henk: For the demo now. This is an HTMX based UI.
+
+Henk: You can click on a property and see history.
+
+Henk: (shows a TD)
+
+Henk: I know that this is not a compliant TD. The forms are omitted.
+
+Henk: Without type annotations, I don't think that the standards will go far.
+
+Henk: I was struggling with enums . I am now going with oneOf so that I can add title.
+
+Henk: HiveOT also has a directory. It shows services and Things, based on tags developed for HiveOT.
+
+Henk: Some devices have a lot of properties. Like this multi sensor having 120. The TD is massive and that is without forms.
+
+Henk: So you can ask how we can interact with Things without forms.
+
+Henk: I want to put them but basically you are interacting with all the Things over the hub. So the expressive power of forms is not needed. We can assume.
+
+Henk: The hub is meant for administrator people. You can see running services for example.
+
+Henk: Now onto the challenges.
+
+Henk: It is better to leave your opinions outside. Reading the specs are important.
+
+Henk: I think that some can be addressed to increase adoption.
+
+Henk: There is no relationship between affordances. Like what happens when a property and action have the same names. You are on your own.
+
+Henk: This is needed to solve a problem. For example an action that changes the state, like switching that changes the switch state. A switch can also be toggled manually. So it needs a property for the state.
+
+Henk: To link these, I give the same name. This problem needs to be solved.
+
+Henk: when to choose actions vs properties. There is no guidance so I made my own rules.
+
+Henk: Same for events vs properties. So properties are internal state and events are external states in hiveot.
+
+Henk: Also for results of actions. It is unclear. The spec says it is dependent on the protocol. It is not nice for a hub as it should be generic.
+
+Henk: There are some other small things like multi-line descriptions and comments.
+
+Henk: Also map type would be nice.
+
+### Discussion
+
+Cris: We should definitely collect the challenges. Very valuable for working group.
+
+Yusuf: The `oneOf` solution is nice. How do you handle property values that are dependent on other properties? Did you face such a case?
+
+Henk: Not myself but I can see that happening.
+
+Ege: I see this use case at Siemens.
+
+Yusuf: We should exchange on this.
+
+Cris: Can you elaborate on sharing the SSE?
+
+Henk: The standard assumes a new connection per request. So I define a message envelope. I defined 3 types of envelope. It is quite powerful this way.
```

Comment:
Henk: The standard assumes a new connection per request. So I define a message envelope. I defined 3 types of envelopes for all interaction with all things over a single SSE connection. It is quite powerful this way.

Reply:
```suggestion
Henk: The standard assumes a new connection per request. So I define a message envelope. I defined 3 types of envelopes for all interaction with all things over a single SSE connection. It is quite powerful this way.
```